### PR TITLE
New MIPS Cheat Sheet

### PR DESCRIPTION
**Pull request by KAUTH**
## People to notify

---

Instant Answer Page: https://duck.co/ia/view/mips_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
Hey @KAUTH 

Thanks for this! I'm afraid we're not accept unsolisted cheat sheets at the moment. 

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Comment by KAUTH** [comment]
@pjhampton we were informed that the "programming mission" was over and that we could now create all purpose cheat sheets (besides, the MIPS cheat sheet is a programming cheat sheet that will help some DDG users-programmmers, I know a lot of electrical engineers from my DDG community that have been expecting this). Hope you can reconsider it and reopen the PR, thanks 😉

**Comment by pjhampton** [comment]
Hey @KAUTH! 😄  Ok, I think you are right. 

I think as a community we should strive to prevent overdoing it with cheat sheets. They rarely get triggered and can consume a lot of time to review, deploy etc. Our time is better spent improving existing IAs and plugging holes in the search space.

File: share/goodie/cheat_sheets/json/mips.json [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    "template_type": "code", 
+    "section_order": [
+        "Arithmetic",
+        "Data Transfer",
+        "Logical",
+        "Bitwise shift",
+        "Conditional branch",
+        "Unconditional jump"
+    ],
+    "aliases": [
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/mips.json]: The rest of the cheat sheet looks good to me but I think we should take out these aliases. They seem pretty board and could lead to false positives. Or if you want to keep the array, just include "mips instruction set" only.

Comment (KAUTH) [inline share/goodie/cheat_sheets/json/mips.json]: @tagawa I agree, I removed the other aliases and added the "mips instruction set" only.

**Comment by tagawa** [comment]
Thanks for the quick change @KAUTH - looks good to me! Look out for this going live in a few days, and thanks again for contributing 😃

**Pull request merged**